pkg/gvks/supportedgvks/supportedgvks.go

```diff
@@ -68,34 +67,18 @@ func resources(smLoader *servicemappingloader.ServiceMappingLoader, serviceMetaL
 }
 
 func DirectResources() (map[schema.GroupVersionKind]bool, error) {
-	crds, err := crdloader.LoadCRDs()
-	if err != nil {
-		return nil, fmt.Errorf("error loading crds: %w", err)
-	}
 	handWrittenIAMTypes := make(map[schema.GroupVersionKind]bool)
 	directResources := make(map[schema.GroupVersionKind]bool)
 	for _, gvk := range BasedOnHandwrittenIAMTypes() {
 		handWrittenIAMTypes[gvk] = true
 	}
-	for _, crd := range crds {
-		if crd.ObjectMeta.Labels["cnrm.cloud.google.com/tf2crd"] == "true" {
+	for gvk, metadata := range SupportedGVKs {
+		if metadata.Labels["cnrm.cloud.google.com/tf2crd"] == "true" {
 			continue
 		}
-		if crd.ObjectMeta.Labels["cnrm.cloud.google.com/dcl2crd"] == "true" {
+		if metadata.Labels["cnrm.cloud.google.com/dcl2crd"] == "true" {
 			continue
 		}
-		versions := crd.Spec.Versions
-		highestVersion := k8s.KCCAPIVersionV1Alpha1
```

Comment:
@maqiuyujoyce I am curious why this logic was necessary to filter out only the highest supported direct gvk version? Wouldn't we want to return all supported versions?

Reply:
The highest version was added for reference doc generation: Only GVK containing v1beta1 version will have the reference doc, and the doc is based on version v1beta1.

Adding same GK with different versions should be fine as long as the reference doc generation is not impacted.

Reply:
Ah, cool. Looks like reference doc generation is unaffected because it filters out alpha versions. So I think we are good then.